source/h1_connection.c

```diff
@@ -1066,6 +1094,31 @@ static int s_decoder_on_header(const struct aws_h1_decoded_header *header, void
                 connection->synced_data.new_stream_error_code = AWS_ERROR_HTTP_CONNECTION_CLOSED;
                 aws_h1_connection_unlock_synced_data(connection);
             } /* END CRITICAL SECTION */
+
+            if (connection->base.client_data) {
+                /**
+                 * RFC-9112 section 9.6.
+                 * A client that receives a "close" connection option MUST cease sending
+                 * requests on that connection and close the connection after reading the
+                 * response message containing the "close" connection option.
+                 *
+                 * Mark the stream has no outgoing_message
+                 **/
+                AWS_LOGF_DEBUG(
+                    AWS_LS_HTTP_STREAM,
+                    "id=%p: Request stream receives a \"close\" connection option. Stop writing to the connection "
```

Comment:
```suggestion
                    "id=%p: Received 'Connection: close' header, no more request data will be sent."
```

Reply:
[debatable] wonder if it's worth putting this within an if-block, so that it's not logged when the outgoing message is aready complete?
```
if (!incoming_stream->is_outgoing_message_done) {
    LOGF(... no more request data will be sent);
    incoming_stream->is_outgoing_message_done = true;
}
s_stop(...)
```

Reply:
I think it's worth to be more clear.